lib/whitespace.coffee

```diff
@@ -31,14 +31,26 @@ class Whitespace
         if atom.config.get('whitespace.ensureSingleTrailingNewline', scope: scopeDescriptor)
           @ensureSingleTrailingNewline(editor)
 
+    editorTextInsertedSubscription = editor.onDidInsertText (event) =>
```

Comment:
I'm thinking it's entire functionality of the clearBlankLineOnNewline can be moved into callback for the editor.OnDidInsertText. This saves on the need to pass variables to a function and retrieve the buffer again in the method. Besides, there is no reason for the clearBlankLineOnNewline to be called outside of the class. The refactored version would look like:
```coffeescript
editorTextInsertedSubscription = editor.onDidInsertText (event) ->
  return unless event.text is '\n'
  return unless buffer.isRowBlank(event.range.start.row)

  scopeDescriptor = editor.getRootScopeDescriptor()
  if atom.config.get('whitespace.removeTrailingWhitespace', scope: scopeDescriptor)
    unless atom.config.get('whitespace.ignoreWhitespaceOnlyLines', scope: scopeDescriptor)
      editor.setIndentationForBufferRow(event.range.start.row, 0)
```
I think this version minimizes the impact of running this callback every time sometime inserts text. Especially since now it checks for the blank line before even bothering to use `atom.config.get('...')` twice.